addons/website/models/mixins.py

```diff
@@ -112,6 +112,34 @@ def _default_cover_properties(self):
             "resize_class": "o_half_screen_height",
         }
 
+    def _get_valid_cover_resize_classes(self):
+        return ['o_half_screen_height', 'o_full_screen_height', 'cover_auto']
+
+    def write(self, vals):
+        if 'cover_properties' not in vals:
+            # Not updating cover properties, normal write.
+            return super().write(vals)
+
+        cover_properties = json_scriptsafe.loads(vals['cover_properties'])
+        resize_class = cover_properties.get('resize_class', '').split()
```

Comment:
Minor/Can ignore: 
```suggestion
        resize_classes = cover_properties.get('resize_class', '').split()
```
Seems like the json entry called `resize_class` is not really correct as there can be multiple ones? Misleading, I would name the python variable correctly

Reply:
Debatable but I prefer that plural-names variables are only used for collections (same as JS/HTML convention for classes: `className` is the string with all the classes, the attr `class` is the string with all the classes, etc). In this case, for the python variable, the string is split into an array so indeed, agree, it should be called `resize_classes`, changing that :+1: